jlm/rvsdg/region.cpp

```diff
@@ -515,6 +515,39 @@ Region::ToString(const util::Annotation & annotation, char labelValueSeparator)
   return util::strfmt(annotation.Label(), labelValueSeparator, value);
 }
 
+static size_t
+computeDepth(const Node & node, std::unordered_map<const Node *, size_t> & depthMap)
+{
+  if (depthMap.find(&node) != depthMap.end())
+  {
+    return depthMap[&node];
+  }
+
+  size_t depth = 0;
+  for (auto & input : node.Inputs())
+  {
+    if (const auto owner = TryGetOwnerNode<Node>(*input.origin()))
+    {
+      depth = std::max(depth, computeDepth(*owner, depthMap) + 1);
+    }
+  }
+  depthMap[&node] = depth;
+
+  return depth;
+}
+
+std::unordered_map<const Node *, size_t>
+computeDepthMap(const Region & region)
+{
+  std::unordered_map<const Node *, size_t> depthMap;
+  for (auto & node : region.Nodes())
```

Comment:
see above -- using top-down traverser makes this much more straight

Reply:
Used top-down traverser